src/cpp/session/modules/environment/SessionEnvironment.cpp

```diff
@@ -485,9 +493,25 @@ Error setEnvironment(boost::shared_ptr<int> pContextDepth,
    if (error)
       return error;
 
-   error = setEnvironmentName(*pContextDepth,
-                              *pCurrentContext,
-                              environmentName);
+   s_monitoredPythonModule = std::string();
+   
+   if (s_environmentLanguage == "R")
+   {
+      error = setEnvironmentName(*pContextDepth,
+                                 *pCurrentContext,
+                                 environmentName);
+   }
+   else if (s_environmentLanguage == "Python")
+   {
+      if (environmentName != s_monitoredPythonModule)
+      {
+         s_monitoredPythonModule = environmentName;
+         
+         ClientEvent event(client_events::kEnvironmentRefresh);
+         module_context::enqueClientEvent(event);
+      }
+   }
```

Comment:
Since this is a plain string value, maybe dump a log entry if it's something other than the values we expect? (realize it's totally impossible for it to get into that state _now_...)

Reply:
Good idea -- I've added a simple warning log message.